src/Interpreters/Access/InterpreterExecuteAsQuery.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <Interpreters/IInterpreter.h>
+#include <Parsers/IAST_fwd.h>
+
+
+namespace DB
+{
+class ASTExecuteAsQuery;
+struct RolesOrUsersSet;
+struct User;
+
+class InterpreterExecuteAsQuery : public IInterpreter, WithMutableContext
```

Comment:
`InterpreterExecuteAs` looks a bit better (because it's rather `execute as <user>`, not `execute as <query>`, and `InterpreterExecuteAsUserQuery` is just a bit too long).